compiler/qsc_frontend/src/resolve.rs

```diff
@@ -360,13 +360,22 @@ impl Resolver {
     }
 
     fn bind_type_parameters(&mut self, decl: &CallableDecl) {
+        // ToDo: This seems like the wrong way to get the item_id
```

Comment:
The reason this seems wrong is that we are grabbing items from the names map while we are also populating that same map. More specifically, this function is only ever called while using the With visitor on a package, and this logic requires that the [bind_global_items](https://github.com/microsoft/qsharp/blob/main/compiler/qsc_frontend/src/resolve.rs#L624) function has already been called.
I can see that that is the case for both the [incremental compiler](https://github.com/microsoft/qsharp/blob/main/compiler/qsc_frontend/src/incremental.rs#L198) and for the [full compiler](https://github.com/microsoft/qsharp/blob/main/compiler/qsc_frontend/src/compile.rs#L511).
In short - it seems risky to get the id like this.

Reply:
Hmm... I think this is a safe assumption to make, and will be verified by the panic you have in the else case below. The reason why this is safe is that it's the same assumption that the resolver itself makes. If a callable name is not in the names list then any use of that callable cannot be resolved to an item id, and name resolution breaks down. I don't think we have a stronger way to enforce this at the moment besides exactly the way you are doing it now. If this assumption ever changes, it would have to be accompanied by a pretty significatn refactor of name resolution as a whole, which would include fixing up this function too.

Reply:
removed todo